core/server/data/migrations/versions/4.17/02-add-offers-table.js

```diff
@@ -0,0 +1,17 @@
+const utils = require('../../utils');
+
+module.exports = utils.addTable('offers', {
+    id: {type: 'string', maxlength: 24, nullable: false, primary: true},
+    name: {type: 'string', maxlength: 191, nullable: false},
+    slug: {type: 'string', maxlength: 191, nullable: false, unique: true},
+    product_id: {type: 'string', maxlength: 24, nullable: false},
+    interval: {type: 'string', maxlength: 50, nullable: false, validations: {isIn: [['monthly', 'yearly']]}},
+    currency: {type: 'string', maxlength: 50, nullable: false},
```

Comment:
Percentage based offers won't have a currency - so I think we need to either have this as `nullable` or we store like `'ANY'` or something for the currency - what do you think?

Reply:
Yea i had the thought, and my feeling was we can always just store the currency for the price at the time offer was created irrespective if it was for percentage based or amount based. For percentage based it will simply get ignored in the logic, while for amount based we can use it, but overall the logic can remain the same and we don't need any extra work for either type, wdyt ?

Reply:
Sure we can store it if you think we should ☺️

Reply:
Have updated the `nullable` to `true` in this case as on second thought it does make sense to leave the option in DB to not store anything for it as you said, we can always ensure in logic we save currency for both anyways :)